ladybug/src/test/java/org/frankframework/ibistesttool/filter/GrayBoxTest.java

```diff
@@ -0,0 +1,37 @@
+package org.frankframework.ibistesttool.filter;
+
+import nl.nn.testtool.Checkpoint;
+import nl.nn.testtool.Report;
+import nl.nn.testtool.storage.StorageException;
+
+import org.frankframework.ibistesttool.util.XmlTestStorage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class GrayBoxTest {
+	private static final XmlTestStorage xmlStorage = new XmlTestStorage();
+	private static final String[] expectedResult = {"2#0", "2#24", "2#36", "2#43"};
+
+	@Test
+	public void shouldReturnCorrectCheckpointTest() throws StorageException {
+		File reportFile = new File(Objects.requireNonNull(this.getClass().getClassLoader().getResource("testReport/Pipeline_HelloWorlds.report.xml")).getFile());
+		Report report = xmlStorage.readReportFromFile(reportFile);
+		GrayBox grayBox = new GrayBox();
+		List<Checkpoint> checkpointsAfterFilter = new ArrayList<>();
+		for (Checkpoint checkpoint : report.getCheckpoints()) {
+			if (grayBox.match(report, checkpoint)) {
+				checkpointsAfterFilter.add(checkpoint);
+			}
+		}
+
+		for (int index = 0; index < checkpointsAfterFilter.size(); index++) {
+			assertEquals(expectedResult[index], checkpointsAfterFilter.get(index).getUid());
+		}
```

Comment:
If `checkpointsAfterFilter` is empty, the implementation fails, but your test thinks it is fine, because the whole for-loop is not executed at all 😉 . Please add something like `assertEquals(5, checkpointsAfterFilter.size());` before the loop, to validate the list contents makes sense. 

Same for `BlackBoxTest.java`

Reply:
Thank you for the feedback. I forgot about that case. I have added the changes in the latest commit